pkg/gcv/configs/config.go

```diff
@@ -240,11 +240,17 @@ func CategorizeYAMLFile(data []byte, dataSource string) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	if template, valid := asConstraintTemplate(unclassified); valid {
+	if template, valid, err := asConstraintTemplate(unclassified); valid {
```

Comment:
having 3 return values is a bit much. Can we rework this in the following way:
switch unclassified.Group {
  case validConstraintTemplateGroup:
    return asConstraintTemplate(unclassified)
  case validConstraintGroup:
    return asConstraint(unclassified)
  default:
    return ....

then asConstraintTemplat() and asConstrtaint() can return either (*ConstraintTemplate, error) or (*Constraint, error).

Reply:
oops. should have used a code block; the indentation is all messed up now. hopefully it's still readable.

Reply:
Still made sense, and done!